customer/interfaces/graphql/queryresolver.go

```diff
@@ -46,12 +47,14 @@ func (r *CustomerResolver) CommerceCustomerStatus(ctx context.Context) (*dtocust
 // CommerceCustomer resolver the commerce customer
 func (r *CustomerResolver) CommerceCustomer(ctx context.Context) (*dtocustomer.CustomerResult, error) {
 	user, err := r.service.GetForIdentity(ctx, web.RequestFromContext(ctx))
-	if err != nil {
-		return nil, err
+	if errors.Is(err, application.ErrNoIdentity) {
+		return &dtocustomer.CustomerResult{IsLoggedIn: false}, nil
```

Comment:
Simply do `return nil, nil`, to indicate that there is no Customer, but also no error.

Reply:
changed.